packages/alfa-selector/src/selector.ts

```diff
@@ -1086,6 +1091,53 @@ export namespace Selector {
     private constructor() {
       super("focus");
     }
+
+    public matches(element: Element, context: Context): boolean {
+      return context.isFocused(element);
+    }
+  }
+
+  /**
+   * @see https://drafts.csswg.org/selectors/#focus-within-pseudo
+   */
+  export class FocusWithin extends Pseudo.Class {
+    public static of(): FocusWithin {
+      return new FocusWithin();
+    }
+
+    private constructor() {
+      super("focus-within");
+    }
+
+    public matches(element: Element, context: Context): boolean {
+      return (
+        context.isFocused(element) ||
+        element
+          .descendants({ flattened: true })
+          .filter(isElement)
+          .some((element) => context.isFocused(element))
```

Comment:
Shouldn't that be eta-contracted (`some(context.isFocused)`)?

Reply:
That would break the `this` binding of `Context#isFocused`.